// sanity-plugin-cloudinary/dist/components/AssetPreview.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _VideoPlayer = _interopRequireDefault(require("./VideoPlayer"));

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AssetPreview = (_ref) => {
  var value = _ref.value,
      layout = _ref.layout;
  var url = value && (0, _utils.assetUrl)(value);

  if (!value || !url) {
    return null;
  }

  switch (value.resource_type) {
    case 'video':
      return /*#__PURE__*/_react.default.createElement(_VideoPlayer.default, {
        src: url,
        kind: "player"
      });

    default:
      return /*#__PURE__*/_react.default.createElement("img", {
        alt: "preview",
        src: url,
        style: {
          maxWidth: layout === 'default' ? '80px' : '100%',
          height: 'auto'
        }
      });
  }
};

var _default = AssetPreview;
exports.default = _default;
