// sanity-plugin-cloudinary/dist/components/AssetDiff.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _diff = require("@sanity/field/diff");

var _VideoPlayer = _interopRequireDefault(require("./VideoPlayer"));

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CloudinaryDiffPreview = (_ref) => {
  var value = _ref.value;

  if (!value) {
    return null;
  }

  var url = (0, _utils.assetUrl)(value);

  if (value.resource_type === 'video') {
    return /*#__PURE__*/_react.default.createElement("section", {
      style: {
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'space-between'
      }
    }, /*#__PURE__*/_react.default.createElement(_VideoPlayer.default, {
      src: url,
      kind: "diff"
    }));
  }

  return /*#__PURE__*/_react.default.createElement("img", {
    alt: "preview",
    src: url,
    style: {
      maxWidth: '100%',
      height: 'auto'
    }
  });
};

var AssetDiff = (_ref2) => {
  var diff = _ref2.diff,
      schemaType = _ref2.schemaType;
  return /*#__PURE__*/_react.default.createElement(_diff.DiffFromTo, {
    diff: diff,
    schemaType: schemaType,
    previewComponent: CloudinaryDiffPreview
  });
};

var _default = AssetDiff;
exports.default = _default;
