// sanity-plugin-cloudinary/dist/components/VideoPlayer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _video = _interopRequireDefault(require("video.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class VideoPlayer extends _react.default.Component {
  constructor() {
    super(...arguments);

    _defineProperty(this, "videoNode", void 0);

    _defineProperty(this, "player", void 0);
  }

  componentDidMount() {
    var src = this.props.src;
    this.player = (0, _video.default)(this.videoNode, {
      sources: [{
        src
      }],
      controls: true
    });
  }

  componentWillUnmount() {
    if (this.player) {
      this.player.dispose();
    }
  }

  render() {
    var kind = this.props.kind;
    var className = {
      player: 'video-js vjs-16-9 vjs-big-play-centered',
      diff: 'video-js vjs-layout-tiny vjs-fluid'
    };
    var style = {
      position: 'relative'
    };
    return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("link", {
      href: "https://vjs.zencdn.net/7.8.4/video-js.css",
      rel: "stylesheet"
    }), /*#__PURE__*/_react.default.createElement("div", {
      "data-vjs-player": true
    }, /*#__PURE__*/_react.default.createElement("video", {
      onClick: event => event.stopPropagation(),
      style: kind === 'diff' ? style : {},
      className: className[kind],
      ref: node => {
        if (node) {
          this.videoNode = node;
        }
      }
    })));
  }

}

exports.default = VideoPlayer;
