// sanity-plugin-cloudinary/dist/schema/cloudinaryAsset.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _CloudinaryInput = _interopRequireDefault(require("../components/CloudinaryInput"));

var _AssetDiff = _interopRequireDefault(require("../components/AssetDiff"));

var _AssetPreview = _interopRequireDefault(require("../components/AssetPreview"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  type: 'object',
  name: 'cloudinary.asset',
  fields: [{
    type: 'string',
    name: 'public_id'
  }, {
    type: 'string',
    name: 'resource_type' // "image", "?"

  }, {
    type: 'string',
    name: 'type' // "upload", "?"

  }, {
    type: 'string',
    name: 'format' // "jpg"

  }, {
    type: 'number',
    name: 'version'
  }, {
    type: 'url',
    name: 'url'
  }, {
    type: 'url',
    name: 'secure_url'
  }, {
    type: 'number',
    name: 'width'
  }, {
    type: 'number',
    name: 'height'
  }, {
    type: 'number',
    name: 'bytes'
  }, {
    type: 'number',
    name: 'duration' // can be null

  }, {
    type: 'array',
    name: 'tags',
    of: [{
      type: 'string'
    }]
  }, {
    type: 'datetime',
    name: 'created_at'
  }, {
    type: 'array',
    name: 'derived',
    of: [{
      type: 'cloudinary.assetDerived'
    }]
  }, {
    type: 'string',
    name: 'access_mode'
  } // context array of unknown content
  // metadata array of unknown content
  ],
  inputComponent: _CloudinaryInput.default,
  diffComponent: _AssetDiff.default,
  preview: {
    select: {
      url: 'url',
      resource_type: 'resource_type',
      derived: 'derived.0.url'
    },

    prepare(_ref) {
      var url = _ref.url,
          derived = _ref.derived,
          resource_type = _ref.resource_type;
      return {
        resource_type,
        url: derived || url
      };
    },

    component: _AssetPreview.default
  }
};
exports.default = _default;
